crawl-ref/source/util/species-gen.py

```diff
@@ -0,0 +1,682 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+import traceback
+import collections
+
+import yaml  # pip install pyyaml
+
+
+class Species(dict):
+    """Parser for YAML definition files.
+
+    If any YAML content is invalid, the relevant parser function below should
+    raise ValueError.
+    """
+    @staticmethod
+    def from_yaml(s):
+        species = Species()
```

Comment:
This bit seems to be missing some validation, especially with numerical fields.

Reply:
Added a lot of validation in ecfd8d3fde -- there may be a little more I could do